csi/node_server.go

```diff
@@ -3,26 +3,89 @@ package csi
 import (
 	"context"
 	"fmt"
+	"io/ioutil"
+	"os"
 
 	"github.com/Sirupsen/logrus"
 	csi "github.com/container-storage-interface/spec/lib/go/csi/v0"
 	csicommon "github.com/kubernetes-csi/drivers/pkg/csi-common"
+	"github.com/pkg/errors"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/rest"
 	"k8s.io/kubernetes/pkg/util/mount"
 	volumeutil "k8s.io/kubernetes/pkg/volume/util"
+
+	"github.com/rancher/longhorn-manager/datastore"
+	lhclientset "github.com/rancher/longhorn-manager/k8s/pkg/client/clientset/versioned"
+	"github.com/rancher/longhorn-manager/types"
 )
 
 type NodeServer struct {
 	*csicommon.DefaultNodeServer
+	lhClient  lhclientset.Interface
+	namespace string
 }
 
 func NewNodeServer(d *csicommon.CSIDriver) *NodeServer {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		logrus.Warningf("NewNodeServer failed to create in-cluster config: %v", err)
+	}
+
+	lhClient, err := lhclientset.NewForConfig(config)
+	if err != nil {
+		logrus.Warningf("NewNodeServer failed to create Longhorn client: %v", err)
+	}
+
+	namespace := os.Getenv(types.EnvPodNamespace)
+	if namespace == "" {
+		logrus.Warnf("Cannot detect pod namespace, environment variable %v is missing, "+
+			"using default namespace", types.EnvPodNamespace)
+		namespace = corev1.NamespaceDefault
```

Comment:
Thinking about it, we should just error out. Shouldn't fall back to default.

Can you fix https://github.com/rancher/longhorn-manager/blob/master/controller/controller_manager.go#L32 as well? As another commit.